### PR TITLE
[BEAM-7237] Refine SparkRunner exception management to fix ViewTest.testNonSingletonSideInput

### PR DESCRIPTION
**Pull request by iemejia**
Categorize correctly also some NeedsRunner tests and fix some typos.

R: @aromanenko-dev

### CODE REVIEW

**Comment by iemejia** [comment]
Run Spark ValidatesRunner

**Comment by iemejia** [comment]
Run Spark ValidatesRunner

**Comment by iemejia** [comment]
Run Java PreCommit

File: runners/spark/src/main/java/org/apache/beam/runners/spark/translation/SparkCombineFn.java [hunk 1]
```diff
@@ -722,6 +722,8 @@ public PipelineOptions getPipelineOptions() {
           WindowedAccumulator.create(this, toValue, windowingStrategy, windowComparator);
       accumulator.add(value, this);
       return accumulator;
+    } catch (RuntimeException ex) {
+      throw ex;
```
Comment (iemejia) [inline runners/spark/src/main/java/org/apache/beam/runners/spark/translation/SparkCombineFn.java:726]: This is exactly what fixes `ViewTest.testNonSingletonSideInput`. Some internal runners-core stuff throw `RuntimeException`s like `IllegalArgumentException` and the tests expect those, so we should not be re wrapping those in a different kind.

File: runners/spark/2/src/main/java/org/apache/beam/runners/spark/structuredstreaming/translation/batch/ParDoTranslatorBatch.java [hunk 2]
```diff
@@ -140,8 +140,8 @@ public void translateTransform(
             doFnSchemaInformation,
             sideInputMapping);
 
-    MultiOuputCoder multipleOutputCoder =
-        MultiOuputCoder.of(SerializableCoder.of(TupleTag.class), outputCoderMap, windowCoder);
+    MultiOutputCoder multipleOutputCoder =
```
Comment (aromanenko-dev) [inline runners/spark/2/src/main/java/org/apache/beam/runners/spark/structuredstreaming/translation/batch/ParDoTranslatorBatch.java:143]: I'd suggest to rename this variable to `multiOutputCoder` but it's minor thing.

**Review by aromanenko-dev — APPROVED** [review]
LGTM, thanks!

**Pull request merged**